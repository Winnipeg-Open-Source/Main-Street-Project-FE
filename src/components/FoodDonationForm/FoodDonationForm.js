import React, { memo } from 'react';
import PropTypes from 'prop-types';
import Input from 'components/Input';
import Select from 'components/Select';
import TextArea from 'components/TextArea';

function FoodDonationForm ({
    name,
    description,
    quantity,
    retailValue,
    condition,
    notes,
    onChange,
}) {
    return (
        <>
            <Input id='name' name='name' placeholder='Name' value={name} onChange={onChange} />
            <Input id='description' name='description' placeholder='Description' value={description} onChange={onChange} />
            <Input id='quantity' name='quantity' placeholder='Quantity' type='number' value={quantity} onChange={onChange} />
            <Input id='retail-value' name='retailValue' placeholder='Retail Value ($)' type='number' value={retailValue} onChange={onChange} />
            <Select id='condition' name='condition' label='Condition' value={condition} onChange={onChange} data-testid='Condition'>
                <option>Perfect</option>
                <option>Great</option>
                <option>Good</option>
                <option>Average</option>
                <option>Poor</option>
            </Select>
            <TextArea id='notes' name='notes' label='Notes' value={notes} onChange={onChange} />
        </>
    );
}

FoodDonationForm.displayName = 'FoodDonationForm';

FoodDonationForm.propTypes = {
    name: PropTypes.string,
    description: PropTypes.string,
    quantity: PropTypes.number,
    retailValue: PropTypes.number,
    condition: PropTypes.string,
    notes: PropTypes.string,
    onChange: PropTypes.func,
};

export default memo(FoodDonationForm);
