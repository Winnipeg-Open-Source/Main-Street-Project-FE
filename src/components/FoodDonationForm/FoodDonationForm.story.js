import React from 'react';
import { action } from '@storybook/addon-actions';
import Form from 'components/Form';
import FoodDonationForm from './FoodDonationForm';
import { mockLocations } from 'tests/mocks/locations';
import { mockLineItems } from 'tests/mocks/items';

const props = {
    name: 'Strawberries',
    description: 'Palette of Juicy Fruit',
    quantityName: 'Rolls',
    retailValue: 100,
    locations: mockLocations,
    lineItems: mockLineItems,
    notes: 'A bunch of notes',
    onChange: action('on change'),
    onAddLineItemClick: action('on line item clicked'),
    onItemQuantityChange: action('on line item quantity changed'),
};

export const EmptyForm = () => (
    <Form>
        <FoodDonationForm />
    </Form>
);

export const CompletedForm = () => (
    <Form>
        <FoodDonationForm {...props} />
    </Form>
);

export default {
    title: 'Forms|FoodDonationForm',
    component: FoodDonationForm,
};
