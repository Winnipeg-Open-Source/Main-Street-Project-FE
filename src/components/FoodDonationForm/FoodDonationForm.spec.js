import React from 'react';
import { render } from '@testing-library/react';
import FoodDonationForm from './FoodDonationForm';

const props = {
    name: 'Strawberries',
    description: 'Palette of Juicy Fruit',
    quantity: 5,
    retailValue: 100,
    condition: 'Great',
    notes: 'A bunch of notes',
    onNameChange: () => {},
    onDescriptionChange: () => {},
    onQuantityChange: () => {},
    onRetailValueChange: () => {},
    onConditionChange: () => {},
    onNotesChange: () => {},
};

describe('FoodDonationForm', () => {
    it ('empty has all placeholders', async () => {
        const { getByLabelText, getByPlaceholderText, getByTestId } = render(<FoodDonationForm />);

        const name = await getByPlaceholderText('Name');
        const description = await getByPlaceholderText('Description');
        const quantity = await getByPlaceholderText('Quantity');
        const retailValue = await getByPlaceholderText('Retail Value ($)');
        const condition = await getByTestId('Condition');
        const notes = await getByLabelText('Notes');

        expect(name.value).toBe('');
        expect(description.value).toBe('');
        expect(quantity.value).toBe('');
        expect(retailValue.value).toBe('');
        expect(condition.value).toBe('Perfect');
        expect(notes.value).toBe('');
    });

    it ('has all fields used', async () => {
        const { getByLabelText, getByPlaceholderText, getByTestId } = render(<FoodDonationForm {...props} />);

        const name = await getByPlaceholderText('Name');
        const description = await getByPlaceholderText('Description');
        const quantity = await getByPlaceholderText('Quantity');
        const retailValue = await getByPlaceholderText('Retail Value ($)');
        const condition = await getByTestId('Condition');
        const notes = await getByLabelText('Notes');

        expect(name.value).toBe('Strawberries');
        expect(description.value).toBe('Palette of Juicy Fruit');
        expect(quantity.value).toBe('5');
        expect(retailValue.value).toBe('100');
        expect(condition.value).toBe('Great');
        expect(notes.value).toBe('A bunch of notes');
    });
});
