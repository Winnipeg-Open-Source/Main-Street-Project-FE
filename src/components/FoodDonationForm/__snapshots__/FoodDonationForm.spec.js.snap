// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FoodDonationForm renders without crashing 1`] = `
<DocumentFragment>
  <input
    class="sc-pTWqp sc-pkfPC eDayKI"
    font-size="2,,1"
    id="name"
    name="name"
    placeholder="Name"
    value=""
  />
  <input
    class="sc-pTWqp sc-pkfPC eDayKI"
    font-size="2,,1"
    id="description"
    name="description"
    placeholder="Description"
    value=""
  />
  <input
    class="sc-pTWqp sc-pkfPC eDayKI"
    font-size="2,,1"
    id="quantity"
    name="quantity"
    placeholder="Quantity"
    type="number"
    value=""
  />
  <input
    class="sc-pTWqp sc-pkfPC eDayKI"
    font-size="2,,1"
    id="retail-value"
    name="retailValue"
    placeholder="Retail Value ($)"
    type="number"
    value=""
  />
  <label
    class="sc-oVpqz cKfBma className"
    color="border.light"
    font-size="10px"
    font-weight="bold"
  >
    Condition
    <div
      class="sc-AxjAm sc-AxiKw iSVrPI"
      width="1"
    >
      <select
        class="sc-pZCuu jeUCDo sc-oTaid sc-psDhf bYPPgQ"
        data-testid="Condition"
        font-size="2,,1"
        id="condition"
        name="condition"
      >
        <option>
          Perfect
        </option>
        <option>
          Great
        </option>
        <option>
          Good
        </option>
        <option>
          Average
        </option>
        <option>
          Poor
        </option>
      </select>
      <svg
        aria-hidden="true"
        class="sc-Axmtr bsYOWx sc-fzqPZZ sc-pRFHb JnCiE"
        color="text.light"
        fill="currentcolor"
        focusable="false"
        height="24"
        role="img"
        tabindex="-1"
        viewBox="0 0 24 24"
        width="24"
      >
        <path
          d="M7.4 8l4.6 4.6L16.6 8 18 9.4l-6 6-6-6L7.4 8z"
        />
      </svg>
    </div>
  </label>
  <label
    class="sc-oVpqz cKfBma"
    color="border.light"
    font-size="10px"
    font-weight="bold"
  >
    Notes
    <textarea
      class="sc-qYhTA sc-qQxFs hnlQXd"
      id="notes"
      name="notes"
    />
  </label>
</DocumentFragment>
`;
